### PR TITLE
TRUNK-5351: Create Diagnosis data access interface.

### PR DESCRIPTION
**Pull request by esirK**
## Description of what I changed

This PR Creates a Diagnosis data access interface

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5351

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ */
+public interface DiagnosisDAO {
+	
+	/**
+	 * Saves the diagnosis
+	 *
+	 * @param diagnosis the diagnosis to save
+	 * @return the saved diagnosis
+	 * @throws DAOException exception thrown if error occurs while saving the diagnosis
+	 */
```
Comment (CeciliaCaroline) [inline api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java]: Well documented method. Looks good

**Comment by pluwum** [comment]
Looks good to me. 👍

**Comment by Arthur236** [comment]
The PR looks good to go.

File: api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	Diagnosis saveDiagnosis(Diagnosis diagnosis) throws DAOException;
+	
+	/**
+	 * Gets a diagnosis
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java]: Gets a diagnosis is not enough description

File: api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	 *
+	 * @param diagnosisId the id of the diagnosis to look for
+	 * @return the diagnosis with the given diagnosisId
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java]: Do not expose the property name diagnosisId to the public. Use diagnosis id

File: api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java (hunk 2 above)
Comment (esirK) [reply api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java]: Updating this

**Comment by dkayiwa** [comment]
@esirK @Arthur236 we somehow deleted this ticket by accident.
@Arthur236 do you mind creating it again like we had?
@esirK can you add a new method called getDiagnosis(Integer diagnosisId) which gets a diagnosis by id?

**Comment by Arthur236** [comment]
Sure thing @dkayiwa

**Comment by esirK** [comment]
@dkayiwa This method already existed with the name `getDiagnosisById()` should I just rename the method??

**Comment by esirK** [comment]
@dkayiwa <https://github.com/dkayiwa> This method already existed with the
name getDiagnosisById() should I just rename the method??

**Comment by dkayiwa** [comment]
Oh i see! Then leave it.

**Pull request merged**